Source/Core/DolphinQt2/Config/ControllersWindow.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2017 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#pragma once
+
+#include <QBoxLayout>
+#include <QCheckBox>
+#include <QComboBox>
+#include <QDialog>
+#include <QFormLayout>
+#include <QGroupBox>
+#include <QLabel>
+#include <QLineEdit>
+#include <QPushButton>
+#include <QRadioButton>
+#include <QVBoxLayout>
+
+#include <unordered_map>
+
+#include "Core/HW/SI/SI.h"
+#include "Core/HW/Wiimote.h"
+
+class ControllersWindow final : public QDialog
+{
+  Q_OBJECT
+public:
+  explicit ControllersWindow(QWidget* parent);
+private slots:
+  void OnWiimoteModeChanged(bool passthrough);
+  void OnWiimoteTypeChanged(int state);
+  void OnGCTypeChanged(int state);
+  void SaveSettings();
+  void UnimplementedButton();
+
+private:
+  void CreateGamecubeLayout();
+  void CreateWiimoteLayout();
+  void CreateAdvancedLayout();
+  void CreateMainLayout();
+  void ConnectWidgets();
+  void LoadSettings();
+
+  // Icons
+  QString m_configure_icon;
+  QString m_gamecube_icon;
+  QString m_wii_icon;
+
+  // Main
+  QVBoxLayout* m_main_layout;
+  QHBoxLayout* m_main_hbox;
+
+  // Gamecube
+  QGroupBox* m_gc_box;
+  QLabel* m_gc_label;
+  QFormLayout* m_gc_layout;
+  QComboBox* m_gc_controller_boxes[4];
+  QPushButton* m_gc_buttons[4];
+  QHBoxLayout* m_gc_groups[4];
+
+  // Wii Mote
+  QGroupBox* m_wiimote_box;
+  QLabel* m_wii_label;
+  QFormLayout* m_wiimote_layout;
+  QComboBox* m_wiimote_boxes[4];
+  QPushButton* m_wiimote_buttons[4];
+  QHBoxLayout* m_wiimote_groups[4];
+  QRadioButton* m_wiimote_emu;
+  QRadioButton* m_wiimote_passthrough;
+  QPushButton* m_wiimote_sync;
+  QPushButton* m_wiimote_reset;
+  QCheckBox* m_wiimote_continous_scanning;
+  QCheckBox* m_wiimote_real_balance_board;
+  QCheckBox* m_wiimote_speaker_data;
+
+  // Advanced
+  QGroupBox* m_advanced_box;
+  QHBoxLayout* m_advanced_layout;
+  QCheckBox* m_advanced_bg_input;
+
+  static const std::unordered_map<SerialInterface::SIDevices, int> m_gc_types;
```

Comment:
Rule of thumb. If it's static and private, and doesn't depend on class state, then it can just be made internally linked in the cpp file instead of being exposed in the header.